src/binding.cc

```diff
@@ -1105,10 +1105,57 @@ size_t v8__ArrayBufferView__CopyContents(const v8::ArrayBufferView& self,
   return ptr_to_local(&self)->CopyContents(dest, byte_length);
 }
 
+struct RustAllocatorVtable {
+  void* (*allocate)(void* handle, size_t length);
+  void* (*allocate_uninitialized)(void* handle, size_t length);
+  void (*free)(void* handle, void* data, size_t length);
+  void* (*reallocate)(void* handle, void* data, size_t old_length,
+                      size_t new_length);
+  void (*drop)(void* handle);
+};
+
+class RustAllocator : public v8::ArrayBuffer::Allocator {
+ private:
+  void* handle;
+  const RustAllocatorVtable* vtable;
+
+ public:
+  RustAllocator(void* handle, const RustAllocatorVtable* vtable) {
+    this->handle = handle;
+    this->vtable = vtable;
+  }
+
+  RustAllocator(const RustAllocator& that) = delete;
+  RustAllocator(RustAllocator&& that) = delete;
+
+  virtual ~RustAllocator() { vtable->drop(handle); }
+
+  virtual void* Allocate(size_t length) {
```

Comment:
I'd mark these `final`:
```suggestion
  void* Allocate(size_t length) final {
```
`final` is only valid on virtual functions so it's equivalent to `virtual void* Allocate(size_t) final override`if you omit the `virtual`.